### PR TITLE
Added Fenwick tree (aka Binary Indexed Tree)

### PR DESCRIPTION
**Pull request by Leefrost**
- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Comments in areas I changed are up to date
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

File: DataStructures/Fenwick/BinaryIndexedTree.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+{
+    /// <summary>
+    /// Represent classical realization of Fenwik tree or Binary Indexed tree.
+    ///
+    /// BITree[0..n] --> Array that represents Binary Indexed Tree.
+    /// arr[0..n-1] --> Input array for which prefix sum is evaluated.
+    /// </summary>
+    public class BinaryIndexedTree
+    {
+        private readonly int[] bITree = new int[1000];
```
Comment (siriak) [inline DataStructures/Fenwick/BinaryIndexedTree.cs:13]: Can't we make the size of that array a parameter and pass it through the constructor?

**Review by siriak — CHANGES_REQUESTED** [review]
Please address my comment and fix issues Codacy found and re-request my review

File: DataStructures/Fenwick/BinaryIndexedTree.cs (hunk 1 above)
Comment (Leefrost) [reply DataStructures/Fenwick/BinaryIndexedTree.cs:13]: Yes, we can. I will update the size of the tree based on the size of the input array.

Comment (Leefrost) [reply DataStructures/Fenwick/BinaryIndexedTree.cs]: Replaced with List. Now tree size can non-fixed.

File: DataStructures/Fenwick/BinaryIndexedTree.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        private readonly List<int> fenwickTree;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BinaryIndexedTree"/> class.
+        /// Create Binary indexed tree from the given array.
+        /// </summary>
+        /// <param name="array">Initial array.</param>
+        public BinaryIndexedTree(int[] array)
+        {
+            fenwickTree = Enumerable.Repeat(0, array.Length + 1).ToList();
```
Comment (siriak) [inline DataStructures/Fenwick/BinaryIndexedTree.cs:23]:
```suggestion
            fenwickTree = new List(array.Length + 1);
```

File: DataStructures/Fenwick/BinaryIndexedTree.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        private readonly List<int> fenwickTree;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BinaryIndexedTree"/> class.
+        /// Create Binary indexed tree from the given array.
+        /// </summary>
+        /// <param name="array">Initial array.</param>
+        public BinaryIndexedTree(int[] array)
+        {
+            fenwickTree = new List(array.Length + 1);
```
Comment (siriak) [inline DataStructures/Fenwick/BinaryIndexedTree.cs:23]:
```suggestion
            fenwickTree = new List<int>(array.Length + 1);
```

File: DataStructures/Fenwick/BinaryIndexedTree.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        private readonly List<int> fenwickTree;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BinaryIndexedTree"/> class.
+        /// Create Binary indexed tree from the given array.
+        /// </summary>
+        /// <param name="array">Initial array.</param>
+        public BinaryIndexedTree(int[] array)
+        {
+            fenwickTree = new List<int>(array.Length + 1);
```
Comment (siriak) [inline DataStructures/Fenwick/BinaryIndexedTree.cs:23]:
```suggestion
        private readonly int[] fenwickTree;

        /// <summary>
        /// Initializes a new instance of the <see cref="BinaryIndexedTree"/> class.
        /// Create Binary indexed tree from the given array.
        /// </summary>
        /// <param name="array">Initial array.</param>
        public BinaryIndexedTree(int[] array)
        {
            fenwickTree = new int[array.Length + 1];
```

**Review by siriak — APPROVED** [review]
LGTM, thanks!

**Pull request merged**